ui/components/component-library/button-base/button-base.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import classnames from 'classnames';
+import { IconName, Icon, IconSize, Text } from '..';
+import {
+  AlignItems,
+  Display,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+  BorderRadius,
+  BackgroundColor,
+  IconColor,
+} from '../../../helpers/constants/design-system';
+import type { PolymorphicRef } from '../box';
+import type { TextProps } from '../text';
+import {
+  ButtonBaseProps,
+  ButtonBaseSize,
+  ButtonBaseComponent,
+} from './button-base.types';
+
+export const ButtonBase: ButtonBaseComponent = React.forwardRef(
+  <C extends React.ElementType = 'button' | 'a'>(
+    {
+      block,
+      children,
+      className = '',
+      href,
+      ellipsis = false,
+      externalLink,
+      size = ButtonBaseSize.Md,
+      startIconName,
+      startIconProps,
+      endIconName,
+      endIconProps,
+      loading,
+      disabled,
+      iconLoadingProps,
+      textProps,
+      color = TextColor.textDefault,
+      iconColor = IconColor.iconDefault,
+      ...props
+    }: ButtonBaseProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    return (
+      <Text
+        as={href ? 'a' : 'button'}
+        backgroundColor={BackgroundColor.backgroundAlternative}
+        variant={TextVariant.bodyMdMedium}
+        color={loading ? TextColor.transparent : color}
+        ref={ref}
+        {...(href ? { href } : {})}
+        {...(disabled && !href ? { disabled } : {})}
+        {...(externalLink
```

Comment:
I don't think this fixes the issue if I do 

`<ButtonBase {...args} size={ButtonBaseSize.Sm} as="a" disabled>`

I can still pass the disabled attribute down to the HTML attribute

<img width="1192" alt="Screenshot 2023-07-20 at 2 30 24 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/e2ba47d8-8c87-44ad-8885-3b8c74540574">

Reply:
I see. I think that is a reasonable assumption then. Although I have seen some a tags with onClicks I don't think that's semantically correct. Should there be some logic that prevents the externalLink prop from adding that HTML attributes for that when it is a button?

<img width="1181" alt="Screenshot 2023-07-20 at 3 31 47 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/38b09622-1bb3-432d-a2c4-81c591e2470a">